### PR TITLE
Add npm publish to GitHub actions

### PR DESCRIPTION
**Pull request by frederikprijck**
### Description

Adds support to publish to npm using GitHub actions

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not the default branch

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+jobs:
+  publish-npm:
+    name: 'NPM'
```
Comment (jonkoops) [inline .github/workflows/publish.yml:14]: This should probably be a more human readable string.
```suggestion
    name: Publish to NPM
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
```
Comment (jonkoops) [inline .github/workflows/publish.yml:26]:
```suggestion
          cache: npm
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: Build package
+        run: npm run build
```
Comment (jonkoops) [inline .github/workflows/publish.yml:29]: Is this step required? Would it not be better to handle this in the `prepublishOnly` script?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 3 above)
Comment (jonkoops) [reply .github/workflows/publish.yml:29]: Cool, then let's just leave it as is :)

**Review by evansims — APPROVED** [review]

**Pull request merged**